manifests/defaults.pp

```diff
@@ -36,9 +43,17 @@
   include haproxy::globals
   $_sort_options_alphabetic = pick($sort_options_alphabetic, $haproxy::globals::sort_options_alphabetic)
 
+  $defaults_options = pick($options, $haproxy::params::defaults_options)
+  if $merge_options {
+    $_defaults_options = $haproxy::params::defaults_options + $defaults_options
+  } else {
+    $_defaults_options = $defaults_options
+    warning("${module_name}: The \$merge_options parameter will default to true in the next major release. Please review the documentation regarding the implications.") # lint:ignore:140chars
```

Comment:
would be nice if we could do the major release soonish to flip the default value.

Reply:
Fun thing: we have the same message (permalink: https://github.com/puppetlabs/puppetlabs-haproxy/blame/a3d28a151723b059c4689d628d60b93b7388c2a5/manifests/config.pp#L28) added in 8adbf88bc353c3882e9c3a5a3b92e20ffa830f26, first released in version 1.4.0 and still here in 7.2.1

Reply:
Instead of adding another warning, we should flip it: https://github.com/puppetlabs/puppetlabs-haproxy/pull/592

Reply:
I suggest to merge #592, then update #588 and remove the warning.

Reply:
@yakatz can you please rebase against main and remove the warning? #592 was merged